### PR TITLE
Optimize calls to `TryGetManifestConfig` in `SbomConfigProvider`

### PR DESCRIPTION
**Pull request by ByAgenT**
I changed getter for `ConfigsDictionary` to exit fast if config dictionary already exists. This getter is a hot path during file hash generation [here](https://github.com/microsoft/sbom-tool/blob/b47f3e1d43c2974a67db1a8ab591f7e29adcaf55/src/Microsoft.Sbom.Api/Executors/FileHasher.cs#L129). Because of that we were calling `TryGetManifestConfig` and `RecordSBOMFormat` methods on every file enumerated, doing unnecessary allocations.

### CODE REVIEW

**Review by edgarrs — APPROVED** [review]

File: src/Microsoft.Sbom.Api/Manifest/Configuration/SbomConfigProvider.cs [hunk 1]
```diff
@@ -73,11 +79,6 @@ public class SbomConfigProvider : ISbomConfigProvider
             ILogger logger,
             IRecorder recorder)
         {
-            if (manifestConfigHandlers is null)
```
Comment (edgarrs) [inline src/Microsoft.Sbom.Api/Manifest/Configuration/SbomConfigProvider.cs:76]: Why is this removed?

Comment (ByAgenT) [reply src/Microsoft.Sbom.Api/Manifest/Configuration/SbomConfigProvider.cs:76]: It is a duplicate null check, there's null check for this right after this statement
```csharp
this.manifestConfigHandlers = manifestConfigHandlers ?? throw new ArgumentNullException(nameof(manifestConfigHandlers));
```
